eks/cluster-loader/clusterloader2/addon.go

```diff
@@ -93,10 +93,19 @@ func (c *ClusterLoader) buildConfigMapData() (map[string]string, error) {
 		bodyString := strings.ReplaceAll(string(body), "\n", "\n    ")
 		dataMap[path.Base(uri)] = bodyString
 	}
+	if len(c.Config.Spec.ClusterLoader.TestOverrides) > 0 {
+		overrides := strings.Join(c.Config.Spec.ClusterLoader.TestOverrides, "\n    ")
+		overrides = strings.ReplaceAll(overrides, "=", ": ")
+		dataMap["overrides.yaml"] = overrides
+	}
 	return dataMap, nil
 }
 
 func (c *ClusterLoader) buildArgs() string {
-	// add report-dir argument for CL2 tests. If one wants to change, check the gotemplate, and look at the job's initContainer's volumes.
-	return fmt.Sprintf("[%s]", strings.Join(append(c.Config.Spec.ClusterLoader.TestParams, "--report-dir=/var/reports/cluster-loader"), ","))
+	// add report-dir and testOverrides for CL2 tests. If one wants to change, check the gotemplate, and look at the job's initContainer's volumes
+	arguments := append(c.Config.Spec.ClusterLoader.TestParams, "--report-dir=/var/reports/cluster-loader")
+	if len(c.Config.Spec.ClusterLoader.TestOverrides) > 0 {
+		arguments = append(arguments, "--testoverrides=/etc/config/overrides.yaml")
```

Comment:
If you always have overrides (and just let it be empty if empty), you don't need an if check here.

Reply:
true, will see if it is fine.

Reply:
Can't have an empty testoverrides.